headless/src/jsMain/kotlin/dev/fritz2/headless/foundation/focus.kt

```diff
@@ -219,25 +219,25 @@ fun Tag<HTMLElement>.trapFocus(restoreFocus: Boolean = true, setInitialFocus: In
 fun Tag<HTMLElement>.trapFocusWhenever(
     condition: Flow<Boolean>,
     restoreFocus: Boolean = true,
-    setInitialFocus: InitialFocus = InitialFocus.TryToSet
+    setInitialFocus: InitialFocus = TryToSet
 ) {
+    condition handledBy {
+        if (it) setInitialFocusOnDemand(setInitialFocus)
```

Comment:
handled by is launching a separate coroutine. If we add the if clause with onEach to the condition-flow in line 228 we could avoid this. Or am I missing something...

Reply:
You are right. I improved the call chain.